src/NuGet.Clients/NuGet.PackageManagement.UI/PackageLoadContext.cs

```diff
@@ -53,57 +55,75 @@ internal class PackageLoadContext
             var frameworks = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
 
             foreach (IProjectContextInfo project in Projects)
             {
-                IProjectMetadataContextInfo projectMetadata = await project.GetMetadataAsync(
-                    ServiceBroker,
-                    CancellationToken.None);
-                NuGetFramework framework = projectMetadata.TargetFramework;
-
-                if (framework != null)
+                if (project.ProjectStyle == ProjectModel.ProjectStyle.PackageReference)
                 {
-                    if (framework.IsAny)
+                    // get the target frameworks for Package Reference style projects
+                    var targetFrameworks = await project.GetTargetFrameworksAsync(ServiceBroker, CancellationToken.None);
+                    foreach (var targetFramework in targetFrameworks)
                     {
-                        // One of the project's target framework is AnyFramework. In this case,
-                        // we don't need to pass the framework filter to the server.
-                        return Enumerable.Empty<string>();
-                    }
-
-                    if (framework.IsSpecificFramework)
-                    {
-                        frameworks.Add(framework.DotNetFrameworkName);
+                        frameworks.Add(targetFramework.ToString());
                     }
                 }
                 else
                 {
-                    // we also need to process SupportedFrameworks
-                    IReadOnlyCollection<NuGetFramework> supportedFrameworks = projectMetadata.SupportedFrameworks;
+                    IProjectMetadataContextInfo projectMetadata = await project.GetMetadataAsync(
+                        ServiceBroker,
+                        CancellationToken.None);
+                    NuGetFramework framework = projectMetadata.TargetFramework;
 
-                    if (supportedFrameworks != null && supportedFrameworks.Count > 0)
+                    if (framework != null)
                     {
-                        foreach (var f in supportedFrameworks)
+                        if (framework.IsAny)
                         {
-                            if (f.IsAny)
+                            // One of the project's target framework is AnyFramework. In this case,
+                            // we don't need to pass the framework filter to the server.
+                            return new List<string>();
+                        }
+
+                        if (framework.IsSpecificFramework)
+                        {
+                            frameworks.Add(framework.ToString());
+                        }
+                    }
+                    else
+                    {
+                        // we also need to process SupportedFrameworks
+                        IReadOnlyCollection<NuGetFramework> supportedFrameworks = projectMetadata.SupportedFrameworks;
```

Comment:
It's just question: What is difference between target framework and supported frame here?

Reply:
I expect this contains supported frameworks in addition to the target framework.